File: VeterinariaLosDogos/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.example.myapplication.fragments.ProfileFragment;
import com.example.myapplication.fragments.RegisterPetFragment;
import com.example.myapplication.fragments.ScheduleFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

/**
 * Activity principal con BottomNavigation:
 * - Perfil (lista mascotas + IoT)
 * - Registrar Mascota
 * - Agendar Cita
 */
public class MainActivity extends AppCompatActivity {

    private BottomNavigationView bottomNav;
    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mAuth = FirebaseAuth.getInstance();

        bottomNav = findViewById(R.id.bottomNav);

        bottomNav.setOnNavigationItemSelectedListener(item -> {
            Fragment fragment;
            int id = item.getItemId();
            if (id == R.id.menu_profile) {
                fragment = new ProfileFragment();
            } else if (id == R.id.menu_register_pet) {
                fragment = new RegisterPetFragment();
            } else if (id == R.id.menu_schedule) {
                fragment = new ScheduleFragment();
            } else {
                return false;
            }

            replaceFragment(fragment);
            return true;
        });

        if (savedInstanceState == null) {
            bottomNav.setSelectedItemId(R.id.menu_profile);
            replaceFragment(new ProfileFragment());
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        FirebaseUser current = mAuth.getCurrentUser();
        if (current == null) {
            startActivity(new Intent(this, LoginActivity.class));
            finish();
        }
    }

    private void replaceFragment(@NonNull Fragment fragment) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragmentContainer, fragment)
                .commit();
    }
}

File: VeterinariaLosDogos/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * Activity principal con BottomNavigation:
 * - Perfil (lista mascotas + IoT)
 * - Registrar Mascota
 * - Agendar Cita
 */
class MainActivity : AppCompatActivity() {
    private var bottomNav: BottomNavigationView? = null
    private var mAuth: FirebaseAuth? = null

    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mAuth = FirebaseAuth.getInstance()

            .also { bottomNav = it }<BottomNavigationView> findViewById (R.id.bottomNav)

        bottomNav.setOnNavigationItemSelectedListener({ item ->
            val fragment: Fragment?
            val id: Int = item.getItemId()
            if (id == R.id.menu_profile) {
                fragment = ProfileFragment()
            } else if (id == R.id.menu_register_pet) {
                fragment = RegisterPetFragment()
            } else if (id == R.id.menu_schedule) {
                fragment = ScheduleFragment()
            } else {
                return@setOnNavigationItemSelectedListener false
            }

            replaceFragment(fragment)
            true
        })

        if (savedInstanceState == null) {
            bottomNav.setSelectedItemId(R.id.menu_profile)
            replaceFragment(ProfileFragment())
        }
    }

    protected override fun onStart() {
        super.onStart()
        val current: FirebaseUser? = mAuth.getCurrentUser()
        if (current == null) {
            startActivity(Intent(this, LoginActivity::class.java))
            finish()
        }
    }

    private fun replaceFragment(fragment: Fragment) {
        getSupportFragmentManager()
            .beginTransaction()
            .replace(R.id.fragmentContainer, fragment)
            .commit()
    }
}
